tests/pass-dep/libc/libc-pipe.rs

```diff
@@ -7,6 +7,8 @@ fn main() {
     test_pipe_threaded();
     test_race();
     test_pipe_array();
+    #[cfg(target_os = "linux")] // No `pipe2` in macos.
```

Comment:
Oh, I didn't realize macOS doesn't have this... then it also needs a check in the shim, similar to [this](https://github.com/rust-lang/miri/blob/124c7adac068f5572a8b4b8f924e764d4a8d0654/src/shims/unix/foreign_items.rs#L671-L676).